consensus/src/simplex/mod.rs

```diff
@@ -330,20 +330,31 @@ mod tests {
 
     const PAGE_SIZE: NonZeroUsize = NZUsize!(1024);
     const PAGE_CACHE_SIZE: NonZeroUsize = NZUsize!(10);
+    const TEST_QUOTA: Quota = Quota::per_second(NZU32!(1_000_000));
 
     /// Register a validator with the oracle.
     async fn register_validator<P: PublicKey>(
-        oracle: &mut Oracle<P>,
+        oracle: &mut Oracle<P, deterministic::Context>,
         validator: P,
+        context: deterministic::Context,
     ) -> (
-        (Sender<P>, Receiver<P>),
-        (Sender<P>, Receiver<P>),
-        (Sender<P>, Receiver<P>),
+        (Sender<P, deterministic::Context>, Receiver<P>),
+        (Sender<P, deterministic::Context>, Receiver<P>),
+        (Sender<P, deterministic::Context>, Receiver<P>),
     ) {
         let mut control = oracle.control(validator.clone());
-        let (vote_sender, vote_receiver) = control.register(0).await.unwrap();
-        let (certificate_sender, certificate_receiver) = control.register(1).await.unwrap();
-        let (resolver_sender, resolver_receiver) = control.register(2).await.unwrap();
+        let (vote_sender, vote_receiver) = control
+            .register(0, TEST_QUOTA, context.clone())
```

Comment:
nit: generally prefer context is first :thinking:

Reply:
I suppose this approach should've led to some changes in the work we just merged on internal channel management.